int16[] case
string dir
---
bool success